### PR TITLE
cascade layers : important

### PR DESCRIPTION
**Pull request by romainmenke**
Implement `!important` handling in `@layer` :

- split rules so that `!important` declarations have their own rule
- assign inverse specificity adjustments to the rules with `!important` declarations

I did a few spot tests in Chrome Canary and transformed CSS with `!important` seems to match native `@layer` styles.

### CODE REVIEW

**Review by oluoluoxenfree — APPROVED** [review]

File: plugins/postcss-cascade-layers/README.md [hunk 1]
```diff
@@ -58,6 +58,62 @@ instructions for:
...
+```pcss
+/* [postcss-cascade-layers]: handling "revert-layer" is unsupported by this plugin and will cause style differences between browser versions. */
+@layer {
+	.foo {
+		color: revert-layer;
+	}
+}
+```
+
+### onMixedLayerOrder
```
Comment (sanajaved7) [inline plugins/postcss-cascade-layers/README.md:83]: @romainmenke what do you think about a different name for this warning? At first glance, I wasn't quite sure what `mixed layer order` meant. Could something like `onMediaLayerOrder` work? Or maybe `onConditionalLayerUse`?

Comment (romainmenke) [reply plugins/postcss-cascade-layers/README.md:83]: I also do not mind if this becomes something longer like : `onConditionalRulesWithLayersInDifferentOrders`

This is not something users will ever have to type, they will copy/paste this from the docs. So I too prefer something that is clear :)

Comment (sanajaved7) [reply plugins/postcss-cascade-layers/README.md:83]: Longer is okay with me too! 😃 My only question is about the `DifferentOrders` part - are we trying to capture that the media query could be changing the layer order? Is something like this clear? `onConditionalRulesChangingLayerOrder`?

Comment (romainmenke) [reply plugins/postcss-cascade-layers/README.md:83]: > are we trying to capture that the media query could be changing the layer order?

yes :)

`onConditionalRulesChangingLayerOrder` this works for me!

Shall I update the code to match this?

Comment (sanajaved7) [reply plugins/postcss-cascade-layers/README.md:83]: @romainmenke awesome and sure! I'm happy to make the change as well if you're busy, either way works!

Comment (romainmenke) [reply plugins/postcss-cascade-layers/README.md:83]: Go ahead :)

Comment (sanajaved7) [reply plugins/postcss-cascade-layers/README.md:83]: @romainmenke would you be able to give me permissions to push to the repo? I'm not able to push up my changes to your branch 😅

Comment (romainmenke) [reply plugins/postcss-cascade-layers/README.md:83]: I think you need to merge this branch first.
Then you will be able to apply the changes to the oddbird repo :)

**Comment by sanajaved7** [comment]
[postCSS plugin](https://trello.com/c/Wj3yjQk4/16-postcss-plugin)

**Pull request merged**